Refresh jazzy documentation

djones6 opened the pull request.

djones6 committed:
Enable jazzy generation

SwiftDevOps committed:
Documentation update [ci skip]

helenmasters reviewed (changes requested):
I think it would be good to make a few improvements to the jazzy doc whilst we're in here!

The description is currently "An implementation of Kitura’s TemplateEngine protocol. In particular this templating engine takes files in Markdown (.md) format and converts them to HTML. In addition this class has some helper methods for taking Markdown formatted text and converting it to HTML.

I think it'd be much clearer to make it something like "An implementation of Kitura’s TemplateEngine protocol that enables a Kitura server to serve HTML content generated from Markdown templates (`.md` files). This class also has some helper methods that take Markdown formatted text from a Data struct, or String and convert it to HTML."

helenmasters commented:
fileExtension - The file extension of files in the views directory that will be rendered by a particular templating engine.

I think this should be "The file extension of files that will be rendered by the KituraMarkdown template engine." Perhaps we should also add "By default Kitura will search for these in the  ./Views/ directory, you can customize this by setting the `router.viewsPath` property."

helenmasters commented:
Within init() - Create a KituraMarkdown instance needs a full stop on the end.

helenmasters commented:
There is a parameter called `options` which appears in a couple of places which needs a sentence next to it rather than "markdown options".

helenmasters commented:
Every `from` parameter (there are 3 references to this parameter) needs a full stop at the end of the associated description.

helenmasters commented:
The `pageTemplate` parameter has the description "The HTML page template used" which needs a full stop at the end.

helenmasters commented:
There is no difference in the descriptions for:
`public static func render(from: String) -> String` and `public static func render(from: String, pageTemplate: String) -> String` I'd like there to be a difference as otherwise why are we passing in pageTemplate?

helenmasters commented:
The API doc for the struct `MarkdownOptions` could do with some descriptions adding and full stops at the end of sentences and parameter descriptions.

helenmasters commented:
It would be good to add at least one usage example, there is one in the github README for this repo if required.

djones6 commented:
I've addressed some of the review comments.  Still need:
- usage example
- some kind of example of how the HTML page template should be overridden (the sole purpose of `MarkdownOptions`).  This doesn't seem to be covered in any of our guides so I have inferred its usage from the code.

djones6 committed:
Test fixes for jazzy-doc generation

SwiftDevOps committed:
[jazzy-doc] Documentation update

djones6 committed:
Remove test change

SwiftDevOps committed:
[jazzy-doc] Documentation update

djones6 committed:
Add usage examples

SwiftDevOps committed:
[jazzy-doc] Documentation update

djones6 commented:
@helenmasters Review comments addressed.  And I've also (hopefully) solved the jazzy-doc mergeability issue.

djones6 committed:
Address review comments

SwiftDevOps committed:
[jazzy-doc] Documentation update

djones6 committed:
Merge branch 'master' into jazzydoc

SwiftDevOps committed:
[jazzy-doc] Documentation update

helenmasters commented on Sources/KituraMarkdown/KituraMarkdown.swift:
````diff
@@ -19,46 +19,87 @@ import Foundation
+ ### Usage Example:
+
+ ```swift
+ let markdownOptions = MarkdownOptions(pageTemplate: "<!DOCTYPE html><html lang=\"en\"><head><meta charset=\"UTF-8\"></head><body><div><snippetInsertLocation></snippetInsertLocation></div></body></html>"))
````
If you remove the extra ')' at the end of this line the PR looks good to go.

djones6 committed:
Remove spurious closing bracket

SwiftDevOps committed:
[jazzy-doc] Documentation update
